### ISSUE TITLE
Struggling with integrating Panel data

### ISSUE THREAD

**Issue by Voggmo** [biomedbigdata/cyanus#13]
Loading the data failed with the following message:
panel[[panel_cols$channel]] %in% colnames(fs) are not all TRUE

Is there any option as non bioinformatician to extract the panel information from an .fcs file quite easy?
I tried 100 times to manually create an .xls or .csv file and the error code above always appears. 

Best 
mario

**Comment by Voggmo** [comment]
[export_1_IL-17, SSC-H subset.zip](https://github.com/biomedbigdata/cyanus/files/10166832/export_1_IL-17.SSC-H.subset.zip)

This is an example for a .fcs file in my analysis.

**Comment by Voggmo** [comment]
[export_1_IL-17, SSC-H subset.zip](https://github.com/biomedbigdata/cyanus/files/10167248/export_1_IL-17.SSC-H.subset.zip)

Thank you for answering!

Just saw my mistake. Thank you. I uploaded again.

I tried without integrating without panel information. In general this works fine. 
Hovewer, I want to exclude markers (IL17, CD45, viablility dye) from analysis, as I pregated already in this .fcs file. But the clustering is also based on these markers. Therefore I want to assign a "marker_class" to exclude them. You unterstand my issue?

**Comment by JudithBernett** [comment]
Thank you, I could open it now! Yes, I understand - the guessPanel automatically excludes some markers, so in your case, the code I specified just resulted in the following panel: 
![Bildschirmfoto 2022-12-06 um 15 37 24](https://user-images.githubusercontent.com/38618495/205940902-3ade0065-6730-4aff-9eca-389874936dd2.png)
You can see all channel names of your fcs file with the following code: 
```
fs <- flowCore::read.FCS("export_1_IL-17, SSC-H subset.fcs")
colnames(fs)
# or the channel names and the marker names with
pData(flowCore::parameters(fs))
```
![Bildschirmfoto 2022-12-06 um 15 51 05](https://user-images.githubusercontent.com/38618495/205944469-b5800b2f-4d49-4763-b70c-749c6d7bc8f6.png)

As I come more from the Bioinformatics side, I don't really know which of these markers are relevant for you :D 
But you could export this table with: 
```
panel <- data.table(pData(flowCore::parameters(fs)))

# remove unnecessary columns
panel <- panel[, -c('range', 'minRange', 'maxRange')]

# remove empty values
panel <- panel[!is.na(panel$desc)]

# rename columns
colnames(panel) <- c('fcs_colname', 'antigen')

# add marker_class column and set them all to state for now
panel$marker_class <- 'state'

# export
fwrite(panel, "path_to_output.csv")
```
You can then either edit the marker classes manually using a text editor on your computer or on our 'Get Started' page. Manually, you can also delete the unnecessary lines in the panel. Does this work?

**Comment by JudithBernett** [comment]
No problem!
[panel.csv](https://github.com/biomedbigdata/cyanus/files/10167854/panel.csv)
I really hope, it works for now. 
We are working on a way that users can modify the automatically guessed panel in the "Get Started" tab so that all this can be avoided :D 

Best
Judith

**Comment by Voggmo** [comment]
It works! THANK YOU! :)
I always hat some format issues. For the future you can maybe also add a representative image how it sould look alike in the .csv itself (for non-bioinformaticians like me :D )

Best
Mario

**Comment by JudithBernett** [comment]
Glad to hear it! You mean like a download button for the example panel and metadata?

**Comment by quirinmanz** [comment]
Closing this issue since it has been fixed with #16 and is now live
